common/pjpeg.c

```diff
@@ -833,6 +833,12 @@ pjpeg_t *pjpeg_create_from_file(const char *path, uint32_t flags, int *error)
     uint8_t *buf = malloc(buflen);
     fseek(f, 0, SEEK_SET);
     int res = fread(buf, 1, buflen, f);
+
+    int error_code = ferror(f);
+    if (error_code ){
+        printf ("Read failed with error code: %d\n", error_code );
+    }
```

Comment:
I don't think the specific `error_code` is significant here. The documentation only says it's nonzero if there is an error. So this could be:
```cpp
if (ferror(f))
    debug_print("Read failed");
```
Use the new `debug_print` to report errors so they can be silenced in release builds. You can set the `errno` as proposed in https://github.com/AprilRobotics/apriltag/pull/224.

You may also have to use `clearerr` to clear the error.

Reply:
Fixed. You'll have to double check this for me. I don't use the C error handling routines very often.